src/main/java/org/pesmypetcare/webservice/controller/petmanager/PetMedicationRestController.java

```diff
@@ -0,0 +1,132 @@
+public class PetMedicationRestController {
+    @Autowired
+    private MedicationService medicationService;
+    /**
+     * Creates a Medication eaten by a pet on the database.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName Date + name of the Medication
+     * @param Medication The Medication entity that contains the attributes of the Medication
+     */
+    @PostMapping("/{owner}/{petName}/{dateName}")
+    public void createMedication(@PathVariable String owner, @PathVariable String petName, @PathVariable String dateName,
+                           @RequestBody MedicationEntity Medication) {
+        medicationService.createMedication (owner, petName, dateName, Medication);
+    }
+
+    /**
+     * Deletes all the medication with the specified owner and pet from certain date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName date + name of the receipt of the medication
+     */
+    @DeleteMapping("/{owner}/{petName}/{dateName}")
+    void deleteByDateAndName(String owner, String petName, String dateName){
+        medicationService.deleteByDateAndName(owner, petName, dateName);
+    }
+
+    /**
+     * Deletes all the Medications of the specified pet from database.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @DeleteMapping("/{owner}/{petName}")
+    public void deleteAllMedications(@PathVariable String owner, @PathVariable String petName)
+            throws DatabaseAccessException {
+        medicationService.deleteAllMedications(owner, petName);
+    }
+
+    /**
+     * Gets a Medication identified by its pet and date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName Date of the Medication
+     * @return The MedicationEntity identified by the data
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/{dateName}")
+    public MedicationEntity getMedicationData(@PathVariable String owner, @PathVariable String petName, @PathVariable String dateName)
+            throws DatabaseAccessException {
+        return medicationService.getMedicationData(owner, petName, dateName);
+    }
+
+    /**
+     * Gets the data from all the specified Medications from the database identified by its pet.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @return The List containing all the Medications from the pet
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}")
+    public List<Map<List<String>, Object>> getAllMedicationData(@PathVariable String owner, @PathVariable String petName)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 121).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)